### PR TITLE
Spellcheck on CI

### PR DESCRIPTION
**Pull request by ybiquitous**
> Which issue, if any, is this issue related to?

None.

> Is there anything in the PR that needs further explanation?

This PR introduces spellcheck on CI by using `codespell` (written in Python).

See:
- https://github.com/codespell-project/codespell
- https://github.com/codespell-project/actions-codespell

---

I automatically fixed some misspells via the following command:

```shell
codespell --write
```

But currently, there are the following false positives. It seems that fixing these doesn't make sense to me.

```console
$ codespell
./lib/utils/__tests__/isValidIdentifier.test.js:20: te ==> the, be, we, to
./lib/utils/__tests__/isValidIdentifier.test.js:22: te ==> the, be, we, to
./lib/rules/no-irregular-whitespace/__tests__/index.js:48: nin ==> inn, min, bin, nine
./lib/rules/selector-list-comma-newline-after/__tests__/index.js:186: nd ==> and, 2nd
./lib/rules/selector-list-comma-newline-after/__tests__/index.js:329: nd ==> and, 2nd
./lib/rules/selector-attribute-quotes/__tests__/index.js:123: te ==> the, be, we, to
./lib/rules/selector-attribute-quotes/__tests__/index.js:174: te ==> the, be, we, to
./lib/rules/selector-attribute-quotes/__tests__/index.js:279: te ==> the, be, we, to
./lib/rules/selector-attribute-quotes/__tests__/index.js:287: te ==> the, be, we, to
./lib/rules/selector-attribute-quotes/__tests__/index.js:288: te ==> the, be, we, to
./lib/rules/selector-type-case/README.md:79: arent ==> aren't
./lib/rules/selector-type-case/__tests__/index.js:308: arent ==> aren't
```

### CODE REVIEW

**Review by Mouvedia — COMMENTED** [review]
LGTM

File: lib/utils/__tests__/checkInvalidCLIOptions.test.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ describe('checkInvalidCLIOptions', () => {
 Invalid option ${r('"--fi"')}. Did you mean ${c('"--fix"')}?
 Invalid option ${r('"--fixx"')}. Did you mean ${c('"--fix"')}?
 Invalid option ${r('"--aix"')}. Did you mean ${c('"--fix"')}?
-Invalid option ${r('"--conig"')}. Did you mean ${c('"--config"')}?
+Invalid option ${r('"--configg"')}. Did you mean ${c('"--config"')}?
```
[the remark this answers is not in the document]
Comment (ybiquitous) [reply lib/utils/__tests__/checkInvalidCLIOptions.test.js]: Good catch! Reverted via d7d353f2.

**Comment by ybiquitous** [comment]
@Mouvedia Thanks for the review!

**Comment by JounQin** [comment]
@ybiquitous 

Any specific reason for choosing `codespell` over [`cspell`](https://github.com/streetsidesoftware/cspell) which is a pure nodejs project and has a VSCode plugin https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker to integrate?

**Comment by ybiquitous** [comment]
I didn't have a strong opinion, but didn't want to add extra npm dependencies.

**Comment by JounQin** [comment]
> I didn't have a strong opinion, but didn't want to add extra npm dependencies.

It sounds like that you prefer to relying on CI checks only instead of checking locally before committing?

**Comment by ybiquitous** [comment]
Yes. `codespell` produces some false positives now. I think it's too strict for developers.

**Pull request merged**